lib/ocpp/common/websocket/websocket_libwebsockets.cpp

```diff
@@ -936,12 +947,27 @@ void WebsocketTlsTPM::request_write() {
 
 void WebsocketTlsTPM::poll_message(const std::shared_ptr<WebsocketMessage>& msg) {
     std::shared_ptr<ConnectionData> local_data = conn_data;
-    if (local_data != nullptr) {
+
+    if (local_data == nullptr) {
+        EVLOG_warning << "Queueing message over TLS websocket with null local data, failing!";
+        return;
+    } else {
         auto cd_tid = local_data->get_lws_thread_id();
 
         if (std::this_thread::get_id() == cd_tid) {
             EVLOG_AND_THROW(std::runtime_error("Deadlock detected, polling send from client lws thread!"));
         }
+
+        // If we are interupted or finalized
+        if (local_data->is_interupted() || local_data->get_state() == EConnectionState::FINALIZED) {
+            EVLOG_warning << "Trying to poll message to interrupted/finalized state!";
+            return;
+        }
+    }
+
+    if (this->m_is_connected == false) {
+        EVLOG_warning << "Trying to poll message without being connected!";
+        return;
```

Comment:
Couldnt this be moved to top of function?

Reply:
Done.